diff --git a/README.md b/README.md
index d15c9b0..902beec 100644
--- a/README.md
+++ b/README.md
@@ -22,7 +22,7 @@ The API is available in the retail Windows and Mac clients. There are also Linux
     * Library designed for building a scripted bots using the API.
     * [Repository](https://github.com/Blizzard/s2client-api)
 * **StarCraft II Linux Packages**
-    * Self contained headless linux StarCraft II builds.
+    * Self-contained headless linux StarCraft II builds.
     * [Documentation](docs/linux.md)
     * [Download](#downloads)
 * **Maps**
@@ -60,7 +60,7 @@ To access the linux packages, map packs and replay packs, you must agree to the
 
 The files are password protected with the password 'iagreetotheeula'.
 
-**By typing in the password ‘iagreetotheeula’ you agree to be bound by the terms of the [AI and Machine Learning License](http://blzdistsc2-a.akamaihd.net/AI_AND_MACHINE_LEARNING_LICENSE.html)**
+**By typing in the password 'iagreetotheeula' you agree to be bound by the terms of the [AI and Machine Learning License](http://blzdistsc2-a.akamaihd.net/AI_AND_MACHINE_LEARNING_LICENSE.html)**
 
 ## Linux Packages
 
@@ -116,7 +116,7 @@ The default installation directories are:
 
 On Linux, the installation directory is the folder you extracted the linux package into.
 
-The folder structure is the same accross all platforms. However you may need to create some folders if they are missing.
+The folder structure is the same across all platforms. However, you may need to create some folders if they are missing.
 
 Standard folder layout:
 * StarCraft II/
@@ -138,6 +138,6 @@ Standard folder layout:
 
 These define the action mappings from ability ids in the protobuf api to the internals of the game.
 They also define some general ids that combine multiple abilities that have a similar semantic meaning
-(eg various forms of burrow, cancel, lift/land, etc). The `stableid.json` is updated occasionally with
+(e.g. various forms of burrow, cancel, lift/land, etc.). The `stableid.json` is updated occasionally with
 the game, but can also be updated manually by downloading the `stableid.json` from here and placing it
 in the root of your `StarCraft II` directory.
diff --git a/buildinfo/README.md b/buildinfo/README.md
index 95a6f04..75cb782 100644
--- a/buildinfo/README.md
+++ b/buildinfo/README.md
@@ -1,4 +1,4 @@
-This is build meta-data published with release candidate versions of StarCraft II.
+This is build metadata published with release candidate versions of StarCraft II.
 
 The data-hash is required to start the game manually without using the .build.info file.
 
diff --git a/docs/linux.md b/docs/linux.md
index 470d63e..d53e2d5 100644
--- a/docs/linux.md
+++ b/docs/linux.md
@@ -24,7 +24,7 @@ Optional:
 * **verbose**
     * Enables logging of all protocol requests/responses to std::err.
 * **eglpath**
-    * Sets the path the to harware rendering library.
+    * Sets the path the to hardware rendering library.
     * Required for using the rendered interface with hardware rendering
     * Example: /usr/lib/nvidia-384/libEGL.so
 * **osmesapath**
@@ -47,11 +47,11 @@ To report bugs with the linux build or protocol, create a new issue on this repo
 
 ## Battle.net Cache
 
-The game needs all of the original map data in order to play back a replay.
+The game needs all the original map data in order to play back a replay.
 For ladder replays, the Windows and Mac client will automatically download any missing dependencies if they are missing.
 The Linux build however runs in an offline mode, so these dependencies need to be provided manually.
 
-A prepopulated cache is provided for all of the replay packs.
+A prepopulated cache is provided for all the replay packs.
 To install the data, copy the "Battle.net" folder into the root directory of the data package.
 
 The same issue occurs for if you want to play on these ladder maps.
diff --git a/docs/protocol.md b/docs/protocol.md
index 36c4a37..91a9730 100644
--- a/docs/protocol.md
+++ b/docs/protocol.md
@@ -14,7 +14,7 @@ The protocol for the connection is defined in protobuf. [The protocol definition
 
 All communication over the connection is based around **Request** and **Response** messages. Requests are used for controlling the state of the application, retrieving data and controlling gameplay. 
 
-The data sent into the game must be exactly a series of protobuf defined “Request” objects. The data sent back from the game will be exactly a series of protobuf defined “Response” objects, whose type exactly matches the order of incoming “Request” objects.
+The data sent into the game must be exactly a series of protobuf defined "Request" objects. The data sent back from the game will be exactly a series of protobuf defined "Response" objects, whose type exactly matches the order of incoming "Request" objects.
 
 You are allowed to send additional requests before receiving a response to an earlier request. Requests will be queued and processed in received order. Keeping the request queue saturated is best for optimal performance.
 
@@ -46,7 +46,7 @@ There are two supported options for controlling the game speed:
     * There is no restriction on how fast or slow you can step.
 * **Realtime Mode**
     * The game simulation will automatically advance.
-    * Uses the “faster” game speed (22.4 gameloops per second).
+    * Uses the "faster" game speed (22.4 gameloops per second).
 
 When in realtime mode, the Step transition in the above diagram will occur automatically.
 
@@ -64,7 +64,7 @@ When in realtime mode, the Step transition in the above diagram will occur autom
     5. Repeat
 4. Send **RequestQuit** to shut down application.
 
-## Play a bot against the built in AI
+## Play a bot against the built-in AI
 1. Launch one instance of the SC2 client.
 2. Send **RequestCreateGame** with a valid single player map. 
 3. Send **RequestJoinGame** with desired player config.
@@ -81,7 +81,7 @@ When in realtime mode, the Step transition in the above diagram will occur autom
 ## Play two bots against each other
 1. Launch two instances of the SC2 client.
 2. Choose one of the instances to act as the game host.
-3. Send **RequestCreateGame** to the host with a valid multi player map.
+3. Send **RequestCreateGame** to the host with a valid multi-player map.
 4. Send **RequestJoinGame** to both of the clients with the desired player config.
     1. Clients will synchronize between themselves. 
     2. They will send **ResponseJoinGame** when both clients are ready to start playing.
@@ -123,7 +123,7 @@ There are additional interfaces that can be enabled to provide more data:
     * Provides various values for evaluating the performance of a player.
     * Designed for measuring performance of a bot and replay analysis.
 
-Which interfaces are active is set using the “InterfaceOptions” message when you configure a game/replay. You are allowed to have multiple interfaces active at the same time. 
+Which interfaces are active is set using the "InterfaceOptions" message when you configure a game/replay. You are allowed to have multiple interfaces active at the same time.
 
 Observations will contain the same game state represented in each of the enabled interfaces.
 
@@ -152,15 +152,15 @@ This is designed to be the simplest representation of human data that can be con
 
 The full set of functionality of the user interface is represented. However, functionally equivalent parts have been simplified. There are many ways for a human to move the camera, but in this interface it has been simplified to a single action.
 
-The only relevent protocol to this interface is **RequestObservation** and **RequestAction**.
+The only relevant protocol to this interface is **RequestObservation** and **RequestAction**.
 
-The game world and user interface are represented seperately. Both the observation and action have seperate sections for these two components.
+The game world and user interface are represented separately. Both the observation and action have separate sections for these two components.
 
 All positions are based on screen space coordinates. The upper left of the images are (0, 0).
 
 ## Rendered
 
-This exposes the full fidelty rendered frame buffer that a human player sees.
+This exposes the full fidelity rendered frame buffer that a human player sees.
 
 This interface is fully supported on all platforms including Linux. On Linux it supports both hardware and software rendering.
 
@@ -175,15 +175,15 @@ There are two main types of errors in the protocol:
 * Protocol usage errors.
 * Errors processing a request.
 
-Which requests are allowed at any given time depends on the current status. For example, you are not allowed to send **RequestSaveReplay** if the current status is Launched as it isn’t in a game. When this occurs, the game will send back an empty Response message with only the error field populated.
+Which requests are allowed at any given time depends on the current status. For example, you are not allowed to send **RequestSaveReplay** if the current status is Launched as it isn't in a game. When this occurs, the game will send back an empty Response message with only the error field populated.
 
-Many requests have errors that are specific to that type of request. For example, if you send **RequestCreateGame** with a map path that doesn’t exist. When this type of error occurs, the game will send back a Response message with the appropriate response type populated (eg. **ResponseCreateGame**), but with only the error field populated.
+Many requests have errors that are specific to that type of request. For example, if you send **RequestCreateGame** with a map path that doesn't exist. When this type of error occurs, the game will send back a Response message with the appropriate response type populated (e.g. **ResponseCreateGame**), but with only the error field populated.
 
 ## Action Errors
 
 When you execute an action with **RequestAction**, the protocol will report back any cases where the actions failed. These are equivalent to the red text error messages that appear on the left side of the screen while playing. There are two different places where these may be reported in the protocol.
 
-All actions are validated before they are executed. If an action fails initial validation, it will be reported back immediately in **ResponseAction**. For example, if you try to build a structure but didn’t have enough resources.
+All actions are validated before they are executed. If an action fails initial validation, it will be reported back immediately in **ResponseAction**. For example, if you try to build a structure but didn't have enough resources.
 
 Actions can also fail in the process of being executed. For example, if a unit was trying to build a structure at a distant location, but when it arrived there was an enemy unit in the way. These types of late errors are reported back in **ResponseObservation**.
 
@@ -201,9 +201,9 @@ You can retrieve a listing of what maps are locally cached by using **RequestAva
 
 ## Replays
 
-The game needs all of the original map data in order to play back a replay.
+The game needs all the original map data in order to play back a replay.
 
-For ladder replays, all the dependencies will be automatially downloaded. ([Not true for linux](linux.md#battlenet-cache))
+For ladder replays, all the dependencies will be automatically downloaded. ([Not true for linux](linux.md#battlenet-cache))
 
 For offline game replays, the original .SC2Map file must be available to play back the replay. This can cause problems when sharing the replay between machines. The simplest workflow is to always store maps in the standard maps directory. This is done by creating a folder named "Maps" in the game install directory.
 
@@ -233,11 +233,11 @@ The order in which units update is also randomized. This makes it so that if two
 
 ## Inaccuracies of Actions
 
-When processing replays, the actions reported may not exactly line up with what occured in the original game.
+When processing replays, the actions reported may not exactly line up with what occurred in the original game.
 
-The protobuf protocol is slightly different than the internal replay format. This can cause slight inaccuracies when converting between these two formats.
+The protobuf protocol is slightly different from the internal replay format. This can cause slight inaccuracies when converting between these two formats.
 
-For example, feature layers are a top down orthogonal projection where the actual game is an tilted perspective transformation. This means that rectangle selections from the original game may not line up exactly with the reported rectangle selection in feature layer space.
+For example, feature layers are a top-down orthogonal projection where the actual game is a tilted perspective transformation. This means that rectangle selections from the original game may not line up exactly with the reported rectangle selection in feature layer space.
 
 ## APM
 
@@ -252,7 +252,7 @@ Example:
 * For the APM score, every control group recall will be counted.
 * For the EPM score, only the last control group recall will be counted.
  
-Different action type incur a different ammount of APM "score":
+Different action type incur a different amount of APM "score":
 * Command with target = 2
 * Command with no target = 1
 * Selection action = 1
@@ -274,7 +274,7 @@ StarCraft II uses a deterministic game simulation. Replays effectively just cont
  
 This means that to play back the replay deterministically, you need to be running on the exact same version that was used in the original game.
 
-The **Patch Version** (eg. "3.17") corresponds to a specific **Binary Version** and **Data Version**. Patches may update these independently, so they both need to be specified to play back a replay correctly. (eg. There may be multiple data versions that use the same binary version)
+The **Patch Version** (e.g. "3.17") corresponds to a specific **Binary Version** and **Data Version**. Patches may update these independently, so they both need to be specified to play back a replay correctly. (e.g. There may be multiple data versions that use the same binary version)
  
 The **Binary Version** is represented by a **Base Build Number**.
 * On disk, this number is stored in the folder names inside the "Versions" folder.
@@ -282,8 +282,8 @@ The **Binary Version** is represented by a **Base Build Number**.
 * This is represented by having a different "BaseXXXXX" subfolders.
  
 The **Data Version** is represented by a **Version Hash**.
-* On disk, this hash stored inside of the ".build.info" file.
-* This hash refers to data inside of the SC2Data folder.
+* On disk, this hash stored inside the ".build.info" file.
+* This hash refers to data inside the SC2Data folder.
 * This folder can contain data from multiple versions inside the same install.
 * However, you can only have 1 active data version. (The one specified inside the .build.info file)
  
@@ -309,7 +309,7 @@ You can also get the version information of the binary itself using **RequestPin
 
 ## Downloading Data
 
-On Windows/Mac, only the Binary/Data for the latest patch is guarenteed to be downloaded.
+On Windows/Mac, only the Binary/Data for the latest patch is guaranteed to be downloaded.
 
 The solution to this is to launch the newest version of the game, and request it to download the required old data for you. This is exposed in the API with this field: RequestReplayInfo::download_data
 
diff --git a/s2clientprotocol/common.proto b/s2clientprotocol/common.proto
index 081f306..5f4e68c 100644
--- a/s2clientprotocol/common.proto
+++ b/s2clientprotocol/common.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
diff --git a/s2clientprotocol/data.proto b/s2clientprotocol/data.proto
index 1ac5e56..c83a7a6 100644
--- a/s2clientprotocol/data.proto
+++ b/s2clientprotocol/data.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -8,7 +8,7 @@ import "s2clientprotocol/common.proto";
 // May not relevant: queueable (everything is queueable).
 // May not be important: AbilSetId - marine stim, marauder stim.
 // Stuff omitted: transient.
-// Stuff that may be important: cost, range, Alignment, targetfilters.
+// Stuff that may be important: cost, range, Alignment, target filters.
 message AbilityData {
   optional uint32 ability_id = 1;                                 // Stable ID.
   optional string link_name = 2;                                  // Catalog name of the ability.
@@ -19,11 +19,12 @@ message AbilityData {
   optional uint32 remaps_to_ability_id = 7;                       // This ability id may be represented by the given more generic id.
 
   enum Target {
+    INVALID_Target = 0;
     None = 1;                                                     // Does not require a target.
     Point = 2;                                                    // Requires a target position.
     Unit = 3;                                                     // Requires a unit to target. Given by position using feature layers.
     PointOrUnit = 4;                                              // Requires either a target point or target unit.
-    PointOrNone = 5;                                              // Requires either a target point or no target. (eg. building add-ons)
+    PointOrNone = 5;                                              // Requires either a target point or no target. (e.g. building add-ons)
   }
 
   optional bool available = 8;                                    // If true, the ability may be used by this set of mods/map.
@@ -37,6 +38,7 @@ message AbilityData {
 }
 
 enum Attribute {
+  INVALID_Attribute = 0;
   Light = 1;
   Armored = 2;
   Biological = 3;
@@ -57,6 +59,7 @@ message DamageBonus {
 
 message Weapon {
   enum TargetType {
+    INVALID_TargetType = 0;
     Ground = 1;
     Air = 2;
     Any = 3;
@@ -64,7 +67,7 @@ message Weapon {
   optional TargetType type = 1;
   optional float damage = 2;
   repeated DamageBonus damage_bonus = 3;
-  optional uint32 attacks = 4;              // Number of hits per attack. (eg. Colossus has 2 beams)
+  optional uint32 attacks = 4;              // Number of hits per attack. (e.g. Colossus has 2 beams)
   optional float range = 5;
   optional float speed = 6;                 // Time between attacks.
 }
diff --git a/s2clientprotocol/debug.proto b/s2clientprotocol/debug.proto
index 60ae90a..42cf722 100644
--- a/s2clientprotocol/debug.proto
+++ b/s2clientprotocol/debug.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -67,6 +67,7 @@ message DebugSphere {
 }
 
 enum DebugGameState {
+  INVALID_DebugGameState = 0;
   show_map = 1;
   control_enemy = 2;
   food = 3;
@@ -94,6 +95,7 @@ message DebugKillUnit {
 
 message DebugTestProcess {
   enum Test {
+    INVALID_DebugTestProcess_Test = 0;
     hang = 1;
     crash = 2;
     exit = 3;
@@ -108,6 +110,7 @@ message DebugSetScore {
 
 message DebugEndGame {
   enum EndResult {
+    INVALID_EndResult = 0;
     Surrender = 1;                  // Default if nothing is set. The current player admits defeat.
     DeclareVictory = 2;
   }
@@ -116,6 +119,7 @@ message DebugEndGame {
 
 message DebugSetUnitValue {
   enum UnitValue {
+    INVALID_UnitValue = 0;
     Energy = 1;
     Life = 2;
     Shields = 3;
diff --git a/s2clientprotocol/error.proto b/s2clientprotocol/error.proto
index fe223e9..8fc3980 100644
--- a/s2clientprotocol/error.proto
+++ b/s2clientprotocol/error.proto
@@ -1,9 +1,10 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
 enum ActionResult {
+  INVALID_ActionResult = 0;
   Success = 1;
   NotSupported = 2;
   Error = 3;
diff --git a/s2clientprotocol/query.proto b/s2clientprotocol/query.proto
index c4270f6..a93f5d6 100644
--- a/s2clientprotocol/query.proto
+++ b/s2clientprotocol/query.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
diff --git a/s2clientprotocol/raw.proto b/s2clientprotocol/raw.proto
index 1a08e6a..b210933 100644
--- a/s2clientprotocol/raw.proto
+++ b/s2clientprotocol/raw.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -59,6 +59,7 @@ message UnitOrder {
 }
 
 enum DisplayType {
+  INVALID_DisplayType = 0;
   Visible = 1;                            // Fully visible
   Snapshot = 2;                           // Dimmed version of unit left behind after entering fog of war
   Hidden = 3;                             // Fully hidden
@@ -66,6 +67,7 @@ enum DisplayType {
 }
 
 enum Alliance {
+  INVALID_Alliance = 0;
   Self = 1;
   Ally = 2;
   Neutral = 3;
@@ -116,7 +118,7 @@ message Unit {
   optional float radar_range = 32;
 
   optional bool is_selected = 11;
-  optional bool is_on_screen = 12;          // Visible and within the camera frustrum.
+  optional bool is_on_screen = 12;          // Visible and within the camera frustum.
   optional bool is_blip = 13;               // Detected by sensor tower
   optional bool is_powered = 35;
   optional bool is_active = 39;             // Building is training/researching (ie animated).
@@ -147,8 +149,8 @@ message Unit {
   optional int32 ideal_harvesters = 29;
   optional float weapon_cooldown = 30;
   optional uint64 engaged_target_tag = 34;
-  optional int32 buff_duration_remain = 43;  // How long a buff or unit is still around (eg mule, broodling, chronoboost).
-  optional int32 buff_duration_max = 44;     // How long the buff or unit is still around (eg mule, broodling, chronoboost).
+  optional int32 buff_duration_remain = 43;  // How long a buff or unit is still around (e.g. MULE, Broodling, Chrono Boost).
+  optional int32 buff_duration_max = 44;     // How long the buff or unit is still around (e.g. MULE, Broodling, Chrono Boost).
   repeated RallyTarget rally_targets = 45;
 }
 
@@ -163,7 +165,7 @@ message Event {
 
 message Effect {
   optional uint32 effect_id = 1;
-  repeated Point2D pos = 2;                     // Effect may impact multiple locations. (eg. Lurker attack)
+  repeated Point2D pos = 2;                     // Effect may impact multiple locations. (e.g. Lurker attack)
   optional Alliance alliance = 3;
   optional int32 owner = 4;
   optional float radius = 5;
diff --git a/s2clientprotocol/sc2api.proto b/s2clientprotocol/sc2api.proto
index d82db04..488e7e3 100644
--- a/s2clientprotocol/sc2api.proto
+++ b/s2clientprotocol/sc2api.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -24,7 +24,7 @@ import "s2clientprotocol/ui.proto";
 //    2) Designate a host, and Request.create_game with a multiplayer map.
 //    3) Call Request.join on BOTH clients. Join will block until both clients connect.
 //    4) Wait for a response from both clients. They can now play/step.
-//    5) Steps should be syncronized. One client may time out if they are not. Multiple step sizes are ok.
+//    5) Steps should be synchronized. One client may time out if they are not. Multiple step sizes are ok.
 //    4) Call Request.leave at any point or when the game ends. Observations will not be valid after this.
 //
 // States:
@@ -155,6 +155,7 @@ message Response {
 }
 
 enum Status {
+  INVALID_Game_Status = 0;
   launched = 1;                                 // Game has been launch and is not yet doing anything.
   init_game = 2;                                // Create game has been called, and the host is awaiting players.
   in_game = 3;                                  // In a single or multiplayer game.
@@ -191,6 +192,7 @@ message LocalMap {
 
 message ResponseCreateGame {
   enum Error {
+    INVALID_ResponseCreateGame_Error = 0;
     MissingMap = 1;
     InvalidMapPath = 2;
     InvalidMapData = 3;
@@ -231,6 +233,7 @@ message ResponseJoinGame {
   optional uint32 player_id = 1;
 
   enum Error {
+    INVALID_ResponseJoinGame_Error = 0;
     MissingParticipation = 1;
     InvalidObservedPlayerId = 2;
     MissingOptions = 3;
@@ -258,6 +261,7 @@ message RequestRestartGame {
 // The defaultRestartGameLoops is specified to be (1<<18) by default
 message ResponseRestartGame {
   enum Error {
+    INVALID_ResponseRestartGame_Error = 0;
     LaunchError = 1;
   }
   optional Error error = 1;
@@ -282,6 +286,7 @@ message RequestStartReplay {
 
 message ResponseStartReplay {
   enum Error {
+     INVALID_ResponseStartReplay_Error = 0;
     MissingReplay = 1;
     InvalidReplayPath = 2;
     InvalidReplayData = 3;
@@ -301,6 +306,7 @@ message RequestMapCommand {
 
 message ResponseMapCommand {
   enum Error {
+    INVALID_ResponseMapCommand_Error = 0;
     NoTriggerError = 1;
   }
   optional Error error = 1;
@@ -453,6 +459,7 @@ message ResponseReplayInfo {
   optional uint32 base_build = 8;
 
   enum Error {
+    INVALID_ResponseReplayInfo_Error = 0;
     MissingReplay = 1;
     InvalidReplayPath = 2;
     InvalidReplayData = 3;
@@ -484,6 +491,7 @@ message RequestSaveMap {
 
 message ResponseSaveMap {
   enum Error {
+    INVALID_ResponseSaveMap_Error = 0;
     InvalidMapData = 1;
   }
   optional Error error = 1;
@@ -514,6 +522,7 @@ message ResponseDebug {
 //
 
 enum Difficulty {
+  INVALID_Difficulty = 0;
   VeryEasy = 1;
   Easy = 2;
   Medium = 3;
@@ -527,12 +536,14 @@ enum Difficulty {
 }
 
 enum PlayerType {
+  INVALID_PlayerType = 0;
   Participant = 1;
   Computer = 2;
   Observer = 3;
 }
 
 enum AIBuild {
+  INVALID_AIBuild = 0;
   RandomBuild = 1;
   Rush = 2;
   Timing = 3;
@@ -638,6 +649,7 @@ message Action {
 
 message ActionChat {
   enum Channel {
+    INVALID_Channel = 0;
     Broadcast = 1;
     Team = 2;
   }
@@ -681,6 +693,7 @@ message ActionObserverCameraFollowUnits {
 }
 
 enum Alert {
+  INVALID_Alert = 0;
   AlertError = 3;
   AddOnComplete = 4;
   BuildingComplete = 5;
@@ -706,6 +719,7 @@ enum Alert {
 }
 
 enum Result {
+  INVALID_Game_Result = 0;
   Victory = 1;
   Defeat = 2;
   Tie = 3;
diff --git a/s2clientprotocol/score.proto b/s2clientprotocol/score.proto
index 71a0ae9..78d9492 100644
--- a/s2clientprotocol/score.proto
+++ b/s2clientprotocol/score.proto
@@ -1,21 +1,22 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
 message Score {
   enum ScoreType {
-    Curriculum = 1;                // map generated score (from curriculum maps with special scoring)
-    Melee = 2;                     // summation of in-progress and current units/buildings value + minerals + vespene
+    INVALID_ScoreType = 0;
+    Curriculum = 1;                // Map generated score (from curriculum maps with special scoring).
+    Melee = 2;                     // Summation of in-progress and current units/buildings value + minerals + vespene.
   }
 
   optional ScoreType score_type = 6;
-  optional int32 score = 7;                 // Note: check score_type to know whether this is a melee score or curriculum score
+  optional int32 score = 7;                 // Note: Check score_type to know whether this is a melee score or curriculum score.
   optional ScoreDetails score_details = 8;
 }
 
 message CategoryScoreDetails {
-  optional float none = 1; // Used when no other category is configured in game data
+  optional float none = 1; // Used when no other category is configured in game data.
   optional float army = 2;
   optional float economy = 3;
   optional float technology = 4;
@@ -59,7 +60,7 @@ message ScoreDetails {
   // Sum of spent vespene at the moment it is spent. For example, this number is incremented by 50 when a reaper is queued but decremented by 50 if it is canceled.
   optional float spent_vespene = 12;
 
-  // The following entries contains floating point values for the following catgories:
+  // The following entries contains floating point values for the following categories:
   //   none - There is no category defined in game data.
   //   army - This category includes all military units but not workers.
   //   economy - This category contains town halls, supply structures, vespene buildings and workers.
@@ -69,9 +70,9 @@ message ScoreDetails {
   // Sum of food, or supply, utilized in the categories above.
   optional CategoryScoreDetails food_used = 13;
 
-  // Sum of enemies catagories destroyed in minerals.
+  // Sum of enemies categories destroyed in minerals.
   optional CategoryScoreDetails killed_minerals = 14;
-  // Sum of enemies catagories destroyed in vespene.
+  // Sum of enemies categories destroyed in vespene.
   optional CategoryScoreDetails killed_vespene = 15;
 
   //  Sum of lost minerals for the player in each category.
@@ -84,21 +85,21 @@ message ScoreDetails {
   // Sum of the lost vespene via destroying the players own units/buildings.
   optional CategoryScoreDetails friendly_fire_vespene = 19;
 
-  // Sum of used minerals for the player in each category for each existing unit or upgrade. Therefore if a unit died worth 50 mierals this number will be decremented by 50.
+  // Sum of used minerals for the player in each category for each existing unit or upgrade. Therefore if a unit died worth 50 minerals this number will be decremented by 50.
   optional CategoryScoreDetails used_minerals = 20;
   // Sum of used vespene for the player in each category. Therefore if a unit died worth 50 vespene this number will be decremented by 50.
   optional CategoryScoreDetails used_vespene = 21;
 
-  // Sum of used minerals throughout the entire game for each category. Unliked used_minerals, this value is never decremented.
+  // Sum of used minerals throughout the entire game for each category. Unlike used_minerals, this value is never decremented.
   optional CategoryScoreDetails total_used_minerals = 22;
-  // Sum of used vespene throughout the entire game for each category. Unliked used_vespene, this value is never decremented.
+  // Sum of used vespene throughout the entire game for each category. Unlike used_vespene, this value is never decremented.
   optional CategoryScoreDetails total_used_vespene = 23;
 
   // Sum of damage dealt to the player's opponent for each category.
   optional VitalScoreDetails total_damage_dealt = 24;
   // Sum of damage taken by the player for each category.
   optional VitalScoreDetails total_damage_taken = 25;
-  // Sum of health healed by the player. Note that technology can be healed (by queens) or repaired (by scvs).
+  // Sum of health healed by the player. Note that technology can be healed (by queens) or repaired (by SCVs).
   optional VitalScoreDetails total_healed = 26;
 
   // Recent raw APM.
diff --git a/s2clientprotocol/spatial.proto b/s2clientprotocol/spatial.proto
index 1e8b736..8454227 100644
--- a/s2clientprotocol/spatial.proto
+++ b/s2clientprotocol/spatial.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -31,7 +31,7 @@ message FeatureLayers {
   optional ImageData player_relative = 11;          // uint8. See "Alliance" enum in raw.proto. Range: [1, 4] 
   optional ImageData unit_density_aa = 14;          // uint8. Density of units overlapping a pixel, anti-aliased. [0.0, 16.0f] encoded into [0, 255].
   optional ImageData unit_density = 15;             // uint8. Count of units overlapping a pixel.
-  optional ImageData effects = 20;                  // uint8. Visuals of persistent abilities. (eg. Psistorm)
+  optional ImageData effects = 20;                  // uint8. Visuals of persistent abilities. (e.g. Psionic Storm)
   optional ImageData hallucinations = 21;           // 1-bit. Whether the unit here is a hallucination.
   optional ImageData cloaked = 22;                  // 1-bit. Whether the unit here is cloaked. Hidden units will show up too, but with less details in other layers.
   optional ImageData blip = 23;                     // 1-bit. Whether the unit here is a blip.
@@ -101,6 +101,7 @@ message ActionSpatialCameraMove {
 message ActionSpatialUnitSelectionPoint {
   optional PointI selection_screen_coord = 1;
   enum Type {
+    INVALID_ActionSpatialUnitSelectionPoint_Type = 0;
     Select = 1;         // Equivalent to normal click. Changes selection to unit.
     Toggle = 2;         // Equivalent to shift+click. Toggle selection of unit.
     AllType = 3;        // Equivalent to control+click. Selects all units of a given type.
diff --git a/s2clientprotocol/ui.proto b/s2clientprotocol/ui.proto
index d760be5..4ec45bd 100644
--- a/s2clientprotocol/ui.proto
+++ b/s2clientprotocol/ui.proto
@@ -1,5 +1,5 @@
 
-syntax = "proto2";
+syntax = "proto3";
 
 package SC2APIProtocol;
 
@@ -90,6 +90,7 @@ message ActionUI {
 
 message ActionControlGroup {
   enum ControlGroupAction {
+    INVALID_ControlGroupAction = 0;
     Recall = 1;             // Equivalent to number hotkey. Replaces current selection with control group.
     Set = 2;                // Equivalent to Control + number hotkey. Sets control group to current selection.
     Append = 3;             // Equivalent to Shift + number hotkey. Adds current selection into control group.
@@ -113,6 +114,7 @@ message ActionSelectLarva {
 
 message ActionSelectIdleWorker {
   enum Type {
+    INVALID_ActionSelectIdleWorker_Type = 0;
     Set = 1;        // Equivalent to click with no modifiers. Replaces selection with single idle worker.
     Add = 2;        // Equivalent to shift+click. Adds single idle worker to current selection.
     All = 3;        // Equivalent to control+click. Selects all idle workers.
@@ -123,6 +125,7 @@ message ActionSelectIdleWorker {
 
 message ActionMultiPanel {
   enum Type {
+    INVALID_ActionMultiPanel_Type = 0;
     SingleSelect = 1;         // Click on icon
     DeselectUnit = 2;         // Shift Click on icon
     SelectAllOfType = 3;      // Control Click on icon.
diff --git a/samples/replay-api/README.md b/samples/replay-api/README.md
index 7afc199..92af70f 100644
--- a/samples/replay-api/README.md
+++ b/samples/replay-api/README.md
@@ -30,4 +30,4 @@ Once the script is run successfully, any matching replay packs will be downloade
 
 To access StarCraft II replay packs, you must agree to the [AI and Machine Learning License](https://blzdistsc2-a.akamaihd.net/AI_AND_MACHINE_LEARNING_LICENSE.html)  
 The files are password protected with the password 'iagreetotheeula'.  
-**By typing in the password ‘iagreetotheeula’ you agree to be bound by the terms of the [AI and Machine Learning License](https://blzdistsc2-a.akamaihd.net/AI_AND_MACHINE_LEARNING_LICENSE.html)**
+**By typing in the password 'iagreetotheeula' you agree to be bound by the terms of the [AI and Machine Learning License](https://blzdistsc2-a.akamaihd.net/AI_AND_MACHINE_LEARNING_LICENSE.html)**
diff --git a/samples/replay-api/download_replays.py b/samples/replay-api/download_replays.py
index 9ac31dd..00f11b2 100755
--- a/samples/replay-api/download_replays.py
+++ b/samples/replay-api/download_replays.py
@@ -65,7 +65,7 @@ class BnetAPI(object):
                 'Failed to get oauth access token. response={}'.format(response))
 
     def get(self, url, params=None):
-        """Make an autorized get request to the api by url."""
+        """Make an authorized get request to the api by url."""
         params = params or {}
         params['namespace'] = API_NAMESPACE,
         headers = {'Authorization': 'Bearer ' + self._token}
diff --git a/setup.py b/setup.py
index 6ede9af..7e720e6 100755
--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ def proto_files(root):
 
 
 def compile_proto(source, python_out, proto_path):
-  """Invoke Protocol Compiler to generate python from given source .proto."""
+  """Invoke Protocol Compiler to generate Python from given source .proto."""
   if not protoc:
     sys.exit('protoc not found. Is the protobuf-compiler installed?\n')
 
